include/library/spdm_transport_storage_lib.h

```diff
@@ -0,0 +1,281 @@
+/**
+ *  Copyright Notice:
+ *  Copyright 2025 DMTF. All rights reserved.
+ *  License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/libspdm/blob/main/LICENSE.md
+ **/
+
+#ifndef STORAGE_TRANSPORT_LIB_H
+#define STORAGE_TRANSPORT_LIB_H
+
+#include "library/spdm_common_lib.h"
+#include "library/spdm_crypt_lib.h"
+
+#define LIBSPDM_STORAGE_MAX_RANDOM_NUMBER_COUNT 32
+
+/*
+ * SPDM Storage transport binding header for request encoding.
+ * This virtual header is not specific to any particular storage type, i.e
+ * SCSI, NVMe or ATA. Instead, it is used to encode requests (host to controller),
+ * to communicate SPDM storage transport information. This information shall then
+ * be used to generate the storage protocol specific commands.
+ * Refer to the storage specification (NVMe/SCSI/ATA) for storage command block
+ * descriptions.
+ *
+ * This header *shall not* be transmitted as a part of the libspdm
+ * message in the storage command block data buffer, instead, be used ONLY as a
+ * means to generate the storage specific command(s).
+ *
+ * +-----------------+--------+-------------------+
+ * |      TYPE       |Security|      Security     |
+ * |                 |Protocol| Protocol Specific |
+ * +-----------------+--------+-------------------+
+ * |Security Protocol|    1   |         2         |
+ * +-----------------+--------+-------------------+
+ */
+#pragma pack(1)
+typedef struct {
+    uint8_t security_protocol;
+    uint16_t security_protocol_specific;
+} spdm_storage_transport_virtual_header_t;
+#pragma pack()
+
+#define LIBSPDM_STORAGE_TRANSPORT_HEADER_SIZE  (1 + 2)
+#define LIBSPDM_STORAGE_TRANSPORT_TAIL_SIZE    (0)
+
+#define LIBSPDM_STORAGE_CMD_DIRECTION_IF_SEND 0x01
+#define LIBSPDM_STORAGE_CMD_DIRECTION_IF_RECV 0x02
+
+#pragma pack(1)
+typedef struct {
+    uint8_t rsvd1;
+    uint8_t desc_type;
+    uint8_t status;
+    uint8_t rsvd2;
+    uint32_t length;
+    uint32_t offset;
+    uint32_t rsvd3;
+} spdm_storage_secured_message_descriptor;
+#pragma pack()
+
+typedef enum
+{
+    SECURED_MSG_DESCRIPTOR_NVME = 0x01,
+    SECURED_MSG_DESCRIPTOR_SCSI = 0x02,
+    SECURED_MSG_DESCRIPTOR_ATA = 0x03,
+    SECURED_MSG_DESCRIPTOR_SPDM = 0x04,
+    SECURED_MSG_DESCRIPTOR_DATA_BUFFER = 0x40
+} spmd_storage_secured_message_descriptor_t;
+
+typedef enum
+{
+    SECURED_MSG_ENCAPSULATED_STATUS_SUCCESS = 0x00,
+    SECURED_MSG_ENCAPSULATED_STATUS_INVALID_CMD = 0x02
+} spdm_storage_secured_message_encapsulated_status_t;
```

Comment:
If this align to DSP0286, it should be moved to industry_standard/spdm_storage_binding.h

Also it should be started with SPDM_STORAGE_ - e.g. SPDM_STORAGE_SECURED_MSG_xxx

Spec also defines 01, 03, FF. I suggest we define all of them, when you move.